Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs

```diff
@@ -301,6 +302,71 @@ syncCompletedRide ride booking = do
     CallBAP.sendRideCompletedUpdateToBAP booking ride fareParameters
   pure $ Common.RideSyncRes Common.RIDE_COMPLETED "Success. Sent ride completed update to bap"
 
+---------------------------------------------------------------------
+
+multipleRideSync :: ShortId DM.Merchant -> Common.MultipleRideSyncReq -> Flow Common.MultipleRideSyncRes
+multipleRideSync merchantShortId rideSyncReq = do
+  merchant <- findMerchantByShortId merchantShortId
+  let rideIds = map (cast @Common.Ride @DRide.Ride) rideSyncReq.rideIds
+  ridesBookingsZip <- QRide.findAllRidesBookingsByRideId merchant.id rideIds 
+ 
+
+  -- merchant access checking
+  for_ ridesBookingsZip $ \(ride, booking) -> do unless (merchant.id == booking.providerId) $ throwError (RideDoesNotExist ride.id.getId)
```

Comment:
this checking is redundant, because you already filtered rides by merchant in query `QRide.findAllRidesBookingsByRideId`